jwst/resample/resample_spec.py

```diff
@@ -361,9 +361,9 @@ def build_interpolated_output_wcs(self, refmodel=None):
 
             # Compute the wavelength array, trimming NaNs from the ends
             # In many cases, a whole slice is NaNs, so ignore those warnings
-            warnings.simplefilter("ignore")
-            wavelength_array = np.nanmedian(lam, axis=spectral_axis)
-            warnings.resetwarnings()
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", category=RuntimeWarning, message="All-NaN")
```

Comment:
Again, does this mean that the warning about data arrays being filled with NaN's will no longer show up in the processing log messages? That could be useful for users to see.  Unless it's an expected thing in some instances. Looks like it pertains exclusively to the wavelength array that's being used with spectral data.

Reply:
The code comment above seems to indicate that the original developer of this code thought these warnings were expected.  I just trusted them, and did not change the current behavior at all.